### PR TITLE
Fix ceph-volume lvm create fails to detect raid devices

### PR DESCRIPTION
**Pull request by bi-shun**
Signed-off-by: BI SHUN KE <aionshun@livemail.tw>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by bi-shun** [comment]
This failed issue happens in software raid device (mdadm)

File: src/ceph-volume/ceph_volume/util/disk.py [hunk 1]
```diff
@@ -328,8 +328,8 @@ def is_device(dev):
         return False
     # use lsblk first, fall back to using stat
     TYPE = lsblk(dev).get('TYPE')
-    if TYPE:
-        return TYPE == 'disk'
+    if TYPE == 'disk' or TYPE.startswith(('raid')):
```
Comment (andrewschoen) [inline src/ceph-volume/ceph_volume/util/disk.py]: if ``TYPE`` is ``None`` then the ``TYPE.startswith(('raid'))`` with raise an ``AttributeError``. You'll need to keep the ``if TYPE:`` before this line.

**Comment by sebastian-philipp** [comment]
ping?

**Comment by sebastian-philipp** [comment]
cc @jschmid1

**Comment by jschmid1** [comment]
> cc @jschmid1 Is this related to one of the things you had observed?

Not quite, but this fix is surely needed. (when @andrewschoen's comment is addressed)

What I observed just recently is a mdraid device reporting `size=0` but that was not ceph-volume's fault it turns out. The information it got from sysfs was wrong.

**Comment by jan--f** [comment]
jenkins test ceph-volume all

**Comment by jan--f** [comment]
@bi-shun Mind rebasing this PR onto master please?

**Pull request closed without merging**